virl2_client/models/lab.py

```diff
@@ -917,146 +935,144 @@ def sync_states(self) -> None:
 
     @check_stale
     def wait_until_lab_converged(
-        self, max_iterations: Optional[int] = None, wait_time: Optional[int] = None
+        self, max_iterations: int | None = None, wait_time: int | None = None
     ) -> None:
-        """Wait until lab converges."""
+        """
+        Waits until the lab converges.
+
+        :param max_iterations: (optional) The maximum number of iterations to wait.
+        :param wait_time: (optional) The time to wait between iterations.
+        """
         max_iter = (
             self.wait_max_iterations if max_iterations is None else max_iterations
         )
         wait_time = self.wait_time if wait_time is None else wait_time
-        _LOGGER.info("Waiting for lab %s to converge", self._id)
+        _LOGGER.info(f"Waiting for lab {self._id} to converge.")
         for index in range(max_iter):
             converged = self.has_converged()
             if converged:
-                _LOGGER.info("Lab %s has booted", self._id)
+                _LOGGER.info(f"Lab {self._id} has booted.")
                 return
 
             if index % 10 == 0:
                 _LOGGER.info(
-                    "Lab has not converged, attempt %s/%s, waiting...",
-                    index,
-                    max_iter,
+                    f"Lab has not converged, attempt {index}/{max_iter}, waiting..."
                 )
             time.sleep(wait_time)
 
-        msg = "Lab %s has not converged, maximum tries %s exceeded" % (
-            self.id,
-            max_iter,
-        )
+        msg = f"Lab {self.id} has not converged, maximum tries {max_iter} exceeded."
         _LOGGER.error(msg)
-        # after maximum retries are exceeded and lab has not converged
-        # error must be raised - it makes no sense to just log info
-        # and let client fail with something else if wait is explicitly
-        # specified
+        # After maximum retries are exceeded and lab has not converged,
+        # an error must be raised - it makes no sense to just log info
+        # and let the client fail with something else if wait is explicitly
+        # specified.
         raise RuntimeError(msg)
 
     @check_stale
     def has_converged(self) -> bool:
-        url = self.lab_base_url + "/check_if_converged"
+        """
+        Checks whether the lab has converged.
+
+        :returns: True if the lab has converged, False otherwise.
+        """
+        url = self._url_for("check_if_converged")
         converged = self._session.get(url).json()
         return converged
 
     @check_stale
-    def start(self, wait: Optional[bool] = None) -> None:
+    def start(self, wait: bool | None = None) -> None:
+        """
+        Starts all the nodes and links in the lab.
```

Comment:
Technically, it starts all nodes - it starts only links connected to nodes which needed to be started as far a I remember. 
Tomas - should we consider that a bug or feature, from a "start lab" perspective?